client/src/main.ts

```diff
@@ -11,19 +11,44 @@ tsHello();
 // (these are TypeScript interfaces defined in `languages.ts`)
 import * as Langs from './languages';
 
+class MarkdownBlockKind implements Langs.BlockKind {
+  language : string;
+  source : string;
+  constructor(source:string) {
+    this.language = "markdown";
+    this.source = source;
+  }
+}
+
+class MarkdownEditor implements Langs.Editor {
+  create(id:string, block:Langs.BlockKind) {
+    let markdownBlock = <MarkdownBlockKind>block;
```

Comment:
This is casting a value of `Langs.BlockKind` to a value of `MarkdownBlockKind` - we know this works, because the `MarkdownEditor` will only be called on Markdown blocks!